src/Components/Inputs/Select.php

```diff
@@ -58,6 +66,7 @@ public function __construct(
 
         match ($this->dataSource) {
             self::DATA_SRC_VALUES => $this->initSrcValues($additional),
+            self::DATA_SRC_RESOURCE => $this->initSrcResources($additional),
```

Comment:
This requires the `$resourceRegistry` param, which is not being passed.

I've been thinking about the original design I had given to Saood for this and I'm increasingly thinking that having this initialized in the constructor will be an issue. The desired use in CAP is that we'll combine platform-level hard-coded resources for APIs, platform-level DB-driven resources, and org-specific DB resources. Only the first thing in that list can be set up in a provider, so we cannot rely on the framework to resolve the `ResourceRegistry` singleton[^1] since it'll need to be set up per-request for CAP[^2].

[^1]: Which isn't to say it cannot ever be relied upon. I think a good default for the package is to let it do that *if* the `ResourceRegistry` is not passed to this component. There's certainly a "happy path" where people set up a couple hard-coded resources and only need that.
[^2]: Since it needs both Eloquent to be booted, and needs to know which org the submission is for to load the right group(s).

There's a similar pattern used in the `File` component, where we have to add the `StorageInterface`. It has an `UploadInterface`, the presence of which indicates to the `Form` that it needs to call `setStorageInterface()` on the component when it's building the form definition.

I think the process that may work best is:

- Add a `ResourceValues` (or a better name if you can think of one) interface & apply it to the `Select`
- `Form::__construct` takes an *optional* `ResourceRegistry`
    - When not passed, it can use `resolve(ResourceRegistry::class)` to set it
- `Form::processComponentDefinition` detects components implementing `ResourceValues` and calls the appropriate set method w/ the `ResourceRegistry`
- Modify the `validateDynamicForm` macro in `DynamicFormsProvider::boot` to take & pass along `?ResourceRegistry`